crates/uv-resolver/src/lock.rs

```diff
@@ -504,21 +538,17 @@ impl Distribution {
 pub(crate) struct DistributionId {
     pub(crate) name: PackageName,
     pub(crate) version: Version,
-    #[serde(skip_serializing_if = "Option::is_none")]
-    pub(crate) extra: Option<ExtraName>,
     pub(crate) source: Source,
 }
 
 impl DistributionId {
-    fn from_annotated_dist(annotated_dist: &AnnotatedDist) -> DistributionId {
+    pub(crate) fn from_annotated_dist(annotated_dist: &AnnotatedDist) -> DistributionId {
```

Comment:
Not thrilled that I'm making so many things `pub(crate)` here. Should I change `ResolutionGraph::to_lock` into a `impl TryFrom<ResolutionGraph` for `Lock`?

Reply:
Yeah I'm also not totally convinced that `ResolutionGraph` will be necessary in the long run.